diff --git a/modules/IdleMonitors.qml b/modules/IdleMonitors.qml
--- a/modules/IdleMonitors.qml
+++ b/modules/IdleMonitors.qml
@@ -7,15 +7,17 @@
 import Quickshell.Services.UPower
 import Caelestia.Config
 import Caelestia.Services
-import qs.services
+import qs.services as Services
 
 Scope {
     id: root
 
     required property Lock lock
-    readonly property bool hasPlayer: Players.list.some(p => p.isPlaying)
+    readonly property bool hasPlayer: Services.Players.list.some(p => p.isPlaying)
     readonly property bool isCharging: !UPower.onBattery
     readonly property bool enabled: {
+        if (Services.IdleInhibitor.enabled)
+            return false;
         if (GlobalConfig.general.idle.inhibitWhenAudio && hasPlayer)
             return false;
         if (GlobalConfig.general.idle.inhibitWhenCharging && isCharging)
@@ -32,7 +34,7 @@
         else if (action === "unlock")
             lock.lock.locked = false;
         else if (typeof action === "string")
-            Hypr.dispatch(Hypr.usingLua && ["dpms off", "dpms on"].includes(action) ? `hl.dsp.dpms({ action = "${action === "dpms off" ? "disable" : "enable"}" })` : action);
+            Services.Hypr.dispatch(Services.Hypr.usingLua && ["dpms off", "dpms on"].includes(action) ? `hl.dsp.dpms({ action = "${action === "dpms off" ? "disable" : "enable"}" })` : action);
         else if (!SessionManager.exec(action))
             Quickshell.execDetached(action);
     }
